src/front/app/features/challenge/components/ChallengeContainer.js

```diff
@@ -1,11 +1,30 @@
 import { connect } from 'react-redux';
-import { fetchChallenge, resetState, watchChallenge, acceptChallenge } from '../challenge.actions';
+import {
+    fetchChallenge,
+    resetState,
+    watchChallenge,
+    acceptChallenge,
+    getAcceptedChallenges,
+    userReceived,
+    getWishList
+} from '../challenge.actions';
+import { getUser } from '../../main/profile/profile.actions';
 import Challenge from './Challenge';
 
-const mapStateToProps = ({ challenge }) => ({ challenge });
+const mapStateToProps = ({ challenge, auth, user }) => (
+    {
+        user,
+        challenge,
+        userId: auth.get('userId')
+    }
+);
 
 const mapDispatchToProps = dispatch => ({
     getChallenge: id => dispatch(fetchChallenge(id)),
+    getWishList: userId => dispatch(getWishList(userId)),
```

Comment:
@borys-ihnatyev, this is needed to show that the challenge is already being watched by the user